sympy/solvers/ode/systems.py

```diff
@@ -1064,3 +1068,323 @@ def neq_nth_linear_constant_coeff_match(eqs, funcs, t):
         return match
 
     return None
+
+
+def _preprocess_eqs(eqs):
+    processed_eqs = []
+    for eq in eqs:
+        processed_eqs.append(eq if isinstance(eq, Equality) else Eq(eq, 0))
+
+    return processed_eqs
+
+
+# For now, this function returns a simple output, later it will
+# be capable of dividing the system of ODEs into strongly and
+# weakly connected components.
+def _component_division(eqs, funcs):
+    return [[[eqs, funcs]]]
+
+
+# Returns: List of equations
+def _linear_ode_solver(match):
+    t = match['t']
+    funcs = match['func']
+
+    rhs = match.get('rhs', None)
+    A = -match['func_coeff']
+    B = match.get('commutative_antiderivative', None)
+    const_idx = match['const_idx']
+    type_of_equation = match['type_of_equation']
+
+    sol_vector = linodesolve(A, t, b=rhs, B=B,
+                             type=type_of_equation, const_idx=const_idx)
+
+    sol = [Eq(f, s) for f, s in zip(funcs, sol_vector)]
+
+    return sol
+
+
+# Returns: (List of List of equations, const_idx) or None
+def _ode_component_solver(eqs, funcs, t, const_idx=0):
+    match = neq_nth_linear_constant_coeff_match(eqs, funcs, t)
+
+    if match:
+        match['const_idx'] = const_idx
+        match['t'] = t
+
+        if match.get('is_linear', False):
+            return [_linear_ode_solver(match)], const_idx + len(eqs)
+        if match.get('is_implicit', False):
+            canon_eqs = match['canon_eqs']
+            sols = []
+            temp_const_idx = 0
+            for canon_eq in canon_eqs:
+                sol = _ode_component_solver(canon_eq, funcs, t, const_idx=const_idx)
+                if sol is not None:
+                    sol, temp_const_idx = sol
+                    sols += sol
+
+            # Note: This const_idx logic has to be
+            # verified
+            return sols, temp_const_idx
```

Comment:
I don't understand what `const_idx` is doing. I would write this like
```
if match.get('is_implicit', False):
    canon_eqs = match['canon_eqs']
else
    canon_eqs = [eqs]
for canon_eq in canon_eqs:
    if is_linear(canon_eq)...
```
Or something like that...

Reply:
`const_idx` keeps tracks of the constant naming across the sub-systems of ODEs since we don't want the sub-system solutions to that same constants in their solutions.

> Or something like that...

Sure, I would do that soon.